scripts/collect.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Collect file
+
+# Bootstrap
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "${SCRIPTDIR}/config.sh"
+
+# Functions
+function download() {
+    local source="${1}" # URL (http/ftp/gs supported)
+    local destination="${2}" # local file path
+    logi "Attempting to download $source --> $destination"
+    if [[ -z $source || -z $destination ]]; then
+        loge "Download not possible because source or destination are not set"
+    elif [[ "$source" =~ ^http.*|^ftp.* ]]; then
+        download_with_http $source $destination
+    elif [[ "$source" =~ ^gs.* ]]; then
+        download_with_gsutil $source $destination
+    else
+        loge "Unable to identify protocol when retrieving data"
+    fi
+}
+
+# HTTP/FTP Downloader
+function download_with_http() {
```

Comment:
This is in fact just _download_, the next one is *download_gcs* and, in the future, you could have something like *download_s3*, *download_azure* ...
This is usually achieved via a strategy pattern in software engineering, where the factory method decides which download strategy to instantiate, and all the strategies implement the same interface, i.e. _download(URI)_, although in this case, the script is so short that we can live without the strategy pattern.